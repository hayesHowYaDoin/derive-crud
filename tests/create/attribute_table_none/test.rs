use derive_crud::{Create};

#[derive(Create)]
struct Test {
    #[crud_id]
    id: i64,
    name: String,
}

#[tokio::main]
async fn main() {
    Test::create("Debbie".to_string()).await.expect("Failed to create entry");
}
