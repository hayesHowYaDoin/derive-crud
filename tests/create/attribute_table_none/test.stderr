error: A struct must be annotated with `#[crud_table("table_name")]` to specify the database table
 --> tests/create/attribute_table_none/test.rs:4:1
  |
4 | / struct Test {
5 | |     #[crud_id]
6 | |     id: i64,
7 | |     name: String,
8 | | }
  | |_^

error[E0599]: no function or associated item named `create` found for struct `Test` in the current scope
  --> tests/create/attribute_table_none/test.rs:15:19
   |
4  | struct Test {
   | ----------- function or associated item `create` not found for this struct
...
15 |     let _ = Test::create(&pool, "Debbie".to_string()).await;
   |                   ^^^^^^ function or associated item not found in `Test`
